PWGJE/EMCALJetTasks/macros/AddTaskEmcalJet.C

```diff
@@ -87,7 +87,7 @@ AliEmcalJetTask* AddTaskEmcalJet(
     AliMCParticleContainer* mcpartCont = new AliMCParticleContainer(trackName);
     partCont = mcpartCont;
   }
-  else if (trackName == "tracks" || trackName == "Tracks") {
+  else if (trackName == "tracks" || trackName == "Tracks" || trackName == "mytracks") {
```

Comment:
In my opinion, this check should only be for standard track collections. Admittedly, it's a bit annoying to have to create a track container by hand, but I think it would also get unmanageable pretty quickly if we added lots custom track collections here.

However, if this is going to be done, the static function `AliEmcalJetTask::AddTaskEmcalJet(...)` must also be modified.

Reply:
Hi Raymond, in principle my container is supposed to remove EPOS tracks from detector level MC.  For the output "tracks" TClonesArray, I want that the jet finder handles this array as AliTrackContainer. Therefore i had to change also AddTaskEmcalJet.

Reply:
Hi @fkrizek , what I'm trying to say is that I don't think you need to modify AddTaskEmcalJet. As far as I understand from the code and your comment, your task outputs a new TClonesArray named "mytracks" (by default). Is that right? To add that track collection as a track container to AliEmcalJetTask, I'm trying to suggest that you modify the way that you define your jet finder:

```
auto jetFinder = AliEmcalJetTask::AddTaskEmcalJet("mytracks", ...);
// Remove the existing particle container which you don't want.
jetFinder->RemoveParticleContainer(0);
// Add your track collection as a track container.
auto trackContainer = jetFinder->AddTrackContainer("mytracks");
 // Set any necessary options
trackContainer->SetParticlePtCut(0.15);
```

While this is a bit more work for you (and a bit redundant in terms of code), it also means that we don't need to modify a widely shared AddTask for just one task. This type of approach is what we use for the embedding framework. I think it is beneficial both in terms of simplify maintenance and keeping complexity down for shared tasks. What do you think?

Reply:
Hi Raymond,   if this can be done like that, than indeed it is not necessary to change the AddTaskEmclaJet.   I guess that on the lego train one will then do for the jet finder task
__R_ADDTASK_->RemoveParticleContainer(0);
__R_ADDTASK_->AddTrackContainer("mytracks");

If this is so, then i do not insist on changing AddTaskEmcalJet.  How shall we proceed?
Can you push to the master just the Filter task + its steering macro?

Reply:
Hi Filip,

Yeah, that's exactly how it can be done on the LEGO train.

To proceed, you'll need to modify your commit and then push to the master of your fork. Since modifying a commit will change the git history, you'll need to force push it. I added a few more comments below, so you might want to address them at the same time